server/lib/src/server/migrations.rs

```diff
@@ -121,16 +122,42 @@ impl QueryServer {
             }
         }
 
-        // This is the start of domain info related migrations which we will need in future
-        // to handle replication. Due to the access control rework, and the addition of "managed by"
-        // syntax, we need to ensure both node "fence" replication from each other. We do this
-        // by changing domain infos to be incompatible during this phase.
-        let domain_info_version = match write_txn.internal_search_uuid(UUID_DOMAIN_INFO) {
+        // Reload if anything in migrations requires it.
+        write_txn.reload()?;
+
+        // This is what tells us if the domain entry existed before or not.
+        let db_domain_version = match write_txn.internal_search_uuid(UUID_DOMAIN_INFO) {
             Ok(e) => Ok(e.get_ava_single_uint32(Attribute::Version).unwrap_or(0)),
             Err(OperationError::NoMatchingEntries) => Ok(0),
             Err(r) => Err(r),
         }?;
-        admin_debug!(?domain_info_version);
+
+        warn!(?db_domain_version);
```

Comment:
This should be a debug and say what we're doing
```suggestion
        info!(?db_domain_version, "Before setting internal domain info");
```